cfme/utils/appliance/__init__.py

```diff
@@ -2625,8 +2626,24 @@ def __init__(self, appliance=None):
 
 
 class BaseCollection(NavigatableMixin):
-    pass
+    def __new__(cls, *args, **kwargs):
+        if inspect.getargspec(cls.__init__).args[1] != 'appliance':
```

Comment:
what if appliance isn't passed at all?

Reply:
This is the IndexError that I saw that only comes from especially 'dumb' implementation/use